packages/dev/core/src/Rendering/IBLShadows/iblShadowsRenderPipeline.ts

```diff
@@ -0,0 +1,979 @@
+import { Constants } from "../../Engines/constants";
+import { EngineStore } from "../../Engines/engineStore";
+import type { AbstractMesh } from "../../Meshes/abstractMesh";
+import { Matrix, Vector3, Vector4, Quaternion } from "../../Maths/math.vector";
+import { Mesh } from "../../Meshes/mesh";
+import type { Scene } from "../../scene";
+import type { BaseTexture } from "../../Materials/Textures/baseTexture";
+import { Texture } from "../../Materials/Textures/texture";
+import type { PrePassEffectConfiguration } from "../prePassEffectConfiguration";
+import { PrePassRenderer } from "../prePassRenderer";
+import { Logger } from "../../Misc/logger";
+import { _IblShadowsVoxelRenderer } from "./iblShadowsVoxelRenderer";
+import { _IblShadowsVoxelTracingPass } from "./iblShadowsVoxelTracingPass";
+
+import "../../Shaders/postprocess.vertex";
+import "../../Shaders/iblShadowGBufferDebug.fragment";
+import "../../Shaders/iblShadowsCombine.fragment";
+import { PostProcess } from "../../PostProcesses/postProcess";
+import type { PostProcessOptions } from "../../PostProcesses/postProcess";
+import { _IblShadowsImportanceSamplingRenderer } from "./iblShadowsImportanceSamplingRenderer";
+import { _IblShadowsSpatialBlurPass } from "./iblShadowsSpatialBlurPass";
+import { _IblShadowsAccumulationPass } from "./iblShadowsAccumulationPass";
+import { ArcRotateCamera } from "../../Cameras/arcRotateCamera";
+import { FreeCamera } from "../../Cameras/freeCamera";
+import { PostProcessRenderPipeline } from "../../PostProcesses/RenderPipeline/postProcessRenderPipeline";
+import { PostProcessRenderEffect } from "core/PostProcesses/RenderPipeline/postProcessRenderEffect";
+import type { Camera } from "core/Cameras/camera";
+
+class IblShadowsSettings {

```

Comment:
this should be exported as it is used in the ctor of the main class ?

Why not using only interface as it is used as a property bag only ?

Reply:
Done.